move the embedded app example into this repo

kevinhughes27 opened the pull request:
As per the discussion in https://github.com/Shopify/embedded-app-example/pull/40 I've moved the embedded app example to an example directory in here. I was able to test the app locally (but I had to use ngrok in order for the assets to load in chrome).

![image](https://cloud.githubusercontent.com/assets/1965489/13022216/c5d31202-d1ad-11e5-8fa3-bc18be20fc30.png)

I removed the notes about deploying to heroku and made this very focused on just running locally to play around with the sdk.

for review:

@kmcphillips @celsodantas 
cc @dominiquesr @dtKinger

kmcphillips commented:
You can repeatedly tell chrome "Yes, I'm sure, HTTP and mixed content is fine" but it's rarely enough. Using an SSL port forwarding service is easier.

kevinhughes27 commented:
To make this a little easier on reviewers I'll explain how I ported this and then make some inline comments.

I generated a new app using the generator and then copied over the controller and views. I removed some extra folders and files that rails made that we don't need to reduce clutter a bit. Functionally I updated the views to use content_for javascript but otherwise they are the same

kevinhughes27 commented on example/README.md:
```diff
@@ -0,0 +1,35 @@
+
+You will be prompted to install the application and will be redirected to the embedded Shopify environment once installed.
+
+For local development most modern browsers will block mixed content. Since Shopify runs on HTTPS and a local development server does not, the browser will block the contents of the iframe. The you can either explicitly allow mixed content for your session, or use an HTTPS forwarding service.
```
README is pretty much the exact same I just removed anything related to heroku since deploy is covered elsewhere. This example is to showcase the esdk

kevinhughes27 commented on example/app/assets/javascripts/modals.js:
```diff
@@ -0,0 +1,99 @@
+    else
+      ShopifyApp.flashNotice("No result returned")
+  });
+}
```
straight copy

kevinhughes27 commented on example/app/assets/stylesheets/application.css:
```diff
@@ -0,0 +1,83 @@
+td, th {
+  border: 1px solid #ccc;
+  padding: 5px;
+}
```
straight copy

kevinhughes27 commented on example/app/controllers/home_controller.rb:
```diff
@@ -0,0 +1,41 @@
+    @next_page = "/pagination?page=#{ @page + 1 }" if @page < @total_pages
+  end
+
+end
```
straight copy

kevinhughes27 commented on example/app/views/home/breadcrumbs.html.erb:
```diff
@@ -0,0 +1,28 @@
+      });
+    });
+  </script>
+<% end %>
```
wrapped in content_for javascript

kevinhughes27 commented on example/app/views/home/buttons.html.erb:
```diff
@@ -0,0 +1,23 @@
+      });
+    });
+  </script>
+<% end %>
```
wrapped in content_for javascript

kevinhughes27 commented on example/app/views/home/form_page.html.erb:
```diff
@@ -0,0 +1,54 @@
+
+    });
+  </script>
+<% end %>
```
wrapped in content_for javascript

kevinhughes27 commented on example/app/views/home/help.html.erb:
```diff
@@ -0,0 +1,14 @@
+      ShopifyApp.Bar.loadingOff();
+    });
+  </script>
+<% end %>
```
wrapped in content_for javascript

kevinhughes27 commented on example/app/views/home/index.html.erb:
```diff
@@ -0,0 +1,68 @@
+      });
+    });
+  </script>
+<% end %>
```
wrapped in content_for javascript

kevinhughes27 commented on example/app/views/home/pagination.html.erb:
```diff
@@ -0,0 +1,29 @@
+      });
+    });
+</script>
+<% end %>
```
wrapped in content_for javascript

kevinhughes27 commented on example/app/views/home/regular_app_page.html.erb:
```diff
@@ -0,0 +1,32 @@
+      });
+    });
+</script>
+<% end %>
```
wrapped in content_for javascript

kevinhughes27 commented on example/app/views/layouts/_flash_messages.html.erb:
```diff
@@ -0,0 +1,15 @@
+    <% end %>
+  });
+</script>
+<% end %>
```
wrapped in content_for javascript

kmcphillips commented:
This is 10/10. :star: 

Are we going to archive the other repo, or just update the README to reference this and label it as deprecated?

kevinhughes27 commented:
There is 94 stars and it looks like mostly non shopifiers. I say we burn all the code and update the readme to link here

kevinhughes27 commented:
ping @celsodantas!

kevinhughes27 commented on shopify_app.gemspec:
```diff
@@ -19,7 +19,7 @@ Gem::Specification.new do |s|
   s.add_development_dependency('mocha')
 
-  s.files         = `git ls-files`.split("\n")
+  s.files         = `git ls-files`.split("\n").reject { |f| f.match(%r{^(test|example)/}) }
```
@celsodantas made a good point that adding this would increase the gem size. I added a reject block here to keep some files out of the gem when it is built

celsodantas replied:
<#

celsodantas commented:
:+1:

celsodantas commented:
btw, we might want to update the ShopifyApp README to point to this code.

kevinhughes27 commented:
Thats a good point. The embedded app example points here now but this README should ref this as well
